src/Shared/ReuseableStringBuilder.cs

```diff
@@ -30,6 +30,12 @@ internal sealed class ReuseableStringBuilder : IDisposable, OpportunisticIntern.
         /// </summary>
         private StringBuilder _borrowedBuilder;
 
+        /// <summary>
+        /// Profiling showed that the hot code path for large string builder calls first IsOrdinalEqualToStringOfSameLength followed by ExpensiveConvertToString
+        /// when IsOrdinalEqualToStringOfSameLength did return true. We can therefore reduce the costs for large strings by over a factor two. 
+        /// </summary>
+        private string _cachedString;
```

Comment:
When is this cache invalidated? It looks like it saved for the entire builder - which looks wrong. What happens if someone mutates the builder?

Reply:
@davkean: The cached string is nulled out on every append, remove, ... which should be fine. Or is the SB somewhere returned which I did overlook? If that is too risky I could overload equals to return the internally created string to make it method local.

Reply:
I think nulling it out on every mutate is good 👍